x-pack/plugins/security_solution/public/timelines/components/timeline/body/actions/add_note_icon_item.tsx

```diff
@@ -24,20 +25,24 @@ const AddEventNoteActionComponent: React.FC<AddEventNoteActionProps> = ({
   showNotes,
   timelineType,
   toggleShowNotes,
-}) => (
-  <ActionIconItem>
-    <NotesButton
-      ariaLabel={ariaLabel}
-      data-test-subj="add-note"
-      showNotes={showNotes}
-      timelineType={timelineType}
-      toggleShowNotes={toggleShowNotes}
-      toolTip={
-        timelineType === TimelineType.template ? i18n.NOTES_DISABLE_TOOLTIP : i18n.NOTES_TOOLTIP
-      }
-    />
-  </ActionIconItem>
-);
+}) => {
+  const { kibanaSecuritySolutionsPrivileges } = useUserPrivileges();
+  return (
+    <ActionIconItem>
+      <NotesButton
+        ariaLabel={ariaLabel}
+        data-test-subj="add-note"
+        isDisabled={kibanaSecuritySolutionsPrivileges.crud === false}
```

Comment:
is it possible to also validate this server-side?

Reply:
I don't know if we have a way to do that atm, but I think it's worth implementing holistically to some of our backend api's here